internal/api/grpc/settings/v2/settings_converter_test.go

```diff
@@ -83,122 +85,6 @@ func Test_loginSettingsToPb(t *testing.T) {
 	}
 }
 
-func Test_isDefaultToResourceOwnerTypePb(t *testing.T) {
-	type args struct {
-		isDefault bool
-	}
-	tests := []struct {
-		args args
-		want settings.ResourceOwnerType
-	}{
-		{
-			args: args{false},
-			want: settings.ResourceOwnerType_RESOURCE_OWNER_TYPE_ORG,
-		},
-		{
-			args: args{true},
-			want: settings.ResourceOwnerType_RESOURCE_OWNER_TYPE_INSTANCE,
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.want.String(), func(t *testing.T) {
-			got := isDefaultToResourceOwnerTypePb(tt.args.isDefault)
-			assert.Equal(t, tt.want, got)
-		})
-	}
-}
-
-func Test_passkeysTypeToPb(t *testing.T) {
-	type args struct {
-		passwordlessType domain.PasswordlessType
-	}
-	tests := []struct {
-		args args
-		want settings.PasskeysType
-	}{
-		{
-			args: args{domain.PasswordlessTypeNotAllowed},
-			want: settings.PasskeysType_PASSKEYS_TYPE_NOT_ALLOWED,
-		},
-		{
-			args: args{domain.PasswordlessTypeAllowed},
-			want: settings.PasskeysType_PASSKEYS_TYPE_ALLOWED,
-		},
-		{
-			args: args{99},
-			want: settings.PasskeysType_PASSKEYS_TYPE_NOT_ALLOWED,
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.want.String(), func(t *testing.T) {
-			got := passkeysTypeToPb(tt.args.passwordlessType)
-			assert.Equal(t, tt.want, got)
-		})
-	}
-}
-
-func Test_secondFactorTypeToPb(t *testing.T) {
-	type args struct {
-		secondFactorType domain.SecondFactorType
-	}
-	tests := []struct {
-		args args
-		want settings.SecondFactorType
-	}{
-		{
-			args: args{domain.SecondFactorTypeOTP},
-			want: settings.SecondFactorType_SECOND_FACTOR_TYPE_OTP,
-		},
-		{
-			args: args{domain.SecondFactorTypeU2F},
-			want: settings.SecondFactorType_SECOND_FACTOR_TYPE_U2F,
-		},
-		{
-			args: args{domain.SecondFactorTypeUnspecified},
-			want: settings.SecondFactorType_SECOND_FACTOR_TYPE_UNSPECIFIED,
-		},
-		{
-			args: args{99},
-			want: settings.SecondFactorType_SECOND_FACTOR_TYPE_UNSPECIFIED,
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.want.String(), func(t *testing.T) {
-			got := secondFactorTypeToPb(tt.args.secondFactorType)
-			assert.Equal(t, tt.want, got)
-		})
-	}
-}
-
-func Test_multiFactorTypeToPb(t *testing.T) {
-	type args struct {
-		typ domain.MultiFactorType
-	}
-	tests := []struct {
-		args args
-		want settings.MultiFactorType
-	}{
-		{
-			args: args{domain.MultiFactorTypeU2FWithPIN},
-			want: settings.MultiFactorType_MULTI_FACTOR_TYPE_U2F_WITH_VERIFICATION,
-		},
-		{
-			args: args{domain.MultiFactorTypeUnspecified},
-			want: settings.MultiFactorType_MULTI_FACTOR_TYPE_UNSPECIFIED,
-		},
-		{
-			args: args{99},
-			want: settings.MultiFactorType_MULTI_FACTOR_TYPE_UNSPECIFIED,
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.want.String(), func(t *testing.T) {
-			got := multiFactorTypeToPb(tt.args.typ)
-			assert.Equal(t, tt.want, got)
-		})
-	}
-}
-
```

Comment:
These tests have no value, as they use the same algorithm like the code, which is also just a hardcoded mapping.

Reply:
i think this is just for ensuring that all types are correctly returned and unknown ignored

Reply:
and we could handle the transition from OTP to TOTP here as well

Reply:
It is exactly the same algorithm, so it is a pseudo test. This is also true for the TOTP and the unspecified mapping.
I'd be happy with a test that ensures that no type mapping is forgotten and unknown types map to unspecified (I can do that).
Do you see my point?

Reply:
what do you mean by same algorithm? it's a function with an input and an output, so classic unit test.
i'm aware that it feels kind of senseless mapping `domain.A` to `proto.A` but still there might be functions where `proto.A` and `proto.1` must be mapped to `domain.B` or vice versa.
BTW the linter will make sure that every type is used in the switch.

Reply:
By same algorithm I mean youe just map each type to another type manually 1:1. The algorithm is in the coders brain in both the program and the test.

I'll add the tests again, we can discuss it another time 😄